cmd/credentialUtil.go

```diff
@@ -139,17 +139,28 @@ func getBlobCredentialType(ctx context.Context, blobResourceURL string, canBePub
 // 1. If there is cached session OAuth token, indicating using token credential.
 // 2. If there is OAuth token info passed from env var, indicating using token credential. (Note: this is only for testing)
 // 3. Otherwise use shared key.
-func getBlobFSCredentialType() (common.CredentialType, error) {
+func getBlobFSCredentialType(ctx context.Context, blobResourceURL string, standaloneSAS bool) (common.CredentialType, error) {
+	resourceURL, err := url.Parse(blobResourceURL)
+	if err != nil {
+		return common.ECredentialType.Unknown(), err
+	}
+
 	if oAuthTokenExists() {
 		return common.ECredentialType.OAuthToken(), nil
 	}
 
+	sas := azbfs.NewBfsURLParts(*resourceURL).SAS
+
+	if isSASExisted := sas.Signature() != ""; isSASExisted || standaloneSAS {
+		return common.ECredentialType.Anonymous(), nil
+	}
+
 	name := os.Getenv("ACCOUNT_NAME")
```

Comment:
The blob code (above) gives preference to SAS token in the cases where both SAS token and OAuth token exists. I think we should do the same here, both for consistency and because if a SAS token is specified, the user (by specifying it) is explicitly opting in to use it.